spec/address_book_spec.rb

```diff
@@ -109,4 +109,91 @@ def check_entry(entry, expected_name, expected_number, expected_email)
 
   end
 
+
+  context "#binary_search" do
```

Comment:
A minor improvement would be to put *all* of the `binary_search` `it` blocks inside the `#binary_search context` block.